### ISSUE TITLE
Control socket does not exist

### ISSUE THREAD

**Issue by payals** [ansible/ansible#17532]
##### ISSUE TYPE

- Bug Report
##### COMPONENT NAME

ssh connection plugin
##### ANSIBLE VERSION

```
2.0.0.2
```
##### OS / ENVIRONMENT

running - linuc mint 17
managing - ubuntu
##### SUMMARY

ssh connects fine first two steps but fails for one of the hosts in the third step. 

```
<54.198.19.137> ESTABLISH SSH CONNECTION FOR USER: ubuntu
<54.198.19.137> SSH: EXEC ssh -C -vvv -o ControlMaster=auto -o ControlPersist=60s -o KbdInteractiveAuthentication=no -o PreferredAuthentications=gssapi-with-mic,gssapi-keyex,hostbased,publickey -o PasswordAuthentication=no -o User=ubuntu -o ConnectTimeout=10 -o ControlPath=/home/payal/.ansible/cp/ansible-ssh-%h-%p-%r -tt 54.198.19.137 '( umask 22 && mkdir -p "$( echo $HOME/.ansible/tmp/ansible-tmp-1473719455.51-30669686014096 )" && echo "$( echo $HOME/.ansible/tmp/ansible-tmp-1473719455.51-30669686014096 )" )'
fatal: [master]: UNREACHABLE! => {"changed": false, "msg": "ERROR! SSH encountered an unknown error. The output was:\nOpenSSH_6.6.1, OpenSSL 1.0.1f 6 Jan 2014\r\ndebug1: Reading configuration data /home/payal/.ssh/config\r\ndebug1: Reading configuration data /etc/ssh/ssh_config\r\ndebug1: /etc/ssh/ssh_config line 19: Applying options for *\r\ndebug1: auto-mux: Trying existing master\r\ndebug1: Control socket \"/home/payal/.ansible/cp/ansible-ssh-54.198.19.137-22-ubuntu\" does not exist\r\ndebug2: ssh_connect: needpriv 0\r\ndebug1: Connecting to 54.198.19.137 [54.198.19.137] port 22.\r\ndebug2: fd 3 setting O_NONBLOCK\r\ndebug1: connect to address 54.198.19.137 port 22: Connection timed out\r\nssh: connect to host 54.198.19.137 port 22: Connection timed out\r\n", "unreachable": true}
```

**Comment by mscherer** [comment]
I updated your comment, so people can read the things that were wrongly commented.

**Comment by ansibot** [comment]
@payals: Thanks for taking the time to open this issue, but it was filed under the wrong github repository. Please create a new issue at https://github.com/ansible/ansible-modules-core/issues
[click here for bot help](https://github.com/ansible/ansibullbot/blob/master/ISSUE_HELP.md)

**Comment by jctanner** [comment]
@payals sorry for closing this. The bot believed the issue to be about the setup module because of what you typed in the component section.

**Comment by jctanner** [comment]
@payals What is the output from this command ? ...

```
ssh -C -vvv -o ControlMaster=auto -o ControlPersist=60s -o KbdInteractiveAuthentication=no -o PreferredAuthentications=gssapi-with-mic,gssapi-keyex,hostbased,publickey -o PasswordAuthentication=no -o User=ubuntu -o ConnectTimeout=10 -o ControlPath=/home/payal/.ansible/cp/ansible-ssh-%h-%p-%r -tt 54.198.19.137 'whoami'
```

needs_info

**Comment by ansibot** [comment]
@payals ping, this issue is still waiting on your feedback. We will close the issue if you do not respond.
[click here for bot help](https://github.com/ansible/ansibullbot/blob/master/ISSUE_HELP.md)

**Comment by ansibot** [comment]
@payals ping, this issue is still waiting on your feedback. We will close the issue if you do not respond.
[click here for bot help](https://github.com/ansible/ansibullbot/blob/master/ISSUE_HELP.md)

**Comment by ansibot** [comment]
@payals ping, this issue is still waiting on your feedback. We will close the issue if you do not respond.
[click here for bot help](https://github.com/ansible/ansibullbot/blob/master/ISSUE_HELP.md)

**Comment by ansibot** [comment]
@payals closure has been requested for this issue or it has timed out waiting for your response.
[click here for bot help](https://github.com/ansible/ansibullbot/blob/master/ISSUE_HELP.md)

**Comment by glenjarvis-repairpal** [comment]
Although this issue is closed, let me  add this info. I am experiencing a problem with the same symptoms. I believe it may be related to (at a previous time), I cleaned `/tmp` and `$HOME/.ansible` (when debugging a different issue previously). 

I see this message in debug level 1 output:

```Control socket _\"/tmp/snip-rails-22-snip\"_ does not exist```

**Comment by moetemad** [comment]
I encounter this issue once ansible is running for a while, it is like the ssh session expires or something along this line.
